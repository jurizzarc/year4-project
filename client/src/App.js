import React, { useState, useEffect } from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { GlobalStyles } from './theme/GlobalStyles';
import { useTheme } from './theme/useTheme';
import UserContext from './contexts/UserContext';
import axios from 'axios';
import Home from './components/views/Home';
import SignUp from './components/views/Authentication/SignUp';
import SignIn from './components/views/Authentication/SignIn';
import Dashboard from './components/views/User/Dashboard';
import '../src/theme/index.css';

const App = () => {
    const BASE_API_URL = 'http://localhost:4000/users'
    const { theme, themeLoaded } = useTheme();
    const [selectedTheme, setSelectedTheme] = useState(theme);
    const [userData, setUserData] = useState({
        token: undefined,
        user: undefined
    });

    useEffect(() => {
        setSelectedTheme(theme);
    }, [themeLoaded]);

    useEffect(() => {
        const checkLoggedIn = async () => {
            let token = localStorage.getItem('auth-token');
            if (token === null) {
                localStorage.setItem('auth-token', '');
                token = '';
            }

            const tokenRes = await axios.post(
                `${BASE_API_URL}/tokenIsValid`,
                null, 
                { headers: { 'x-auth-token': token } }
            );
            
            if (tokenRes.data) {
                const userRes = await axios.get(
                    `${BASE_API_URL}/userInfo`,
                    { headers: { 'x-auth-token': token } }
                );

                setUserData({
                    token,
                    user: userRes.data
                });
            }
        };

        checkLoggedIn();
    }, []);

    return (
        <>
            <BrowserRouter>
                <UserContext.Provider value={{userData, setUserData}}>
                    {
                        themeLoaded && <ThemeProvider theme={selectedTheme}>
                            <GlobalStyles />
                            <Switch>
                                <Route
                                    exact path="/"
                                    component={Home}
                                />
                                <Route 
                                    path="/sign-up" 
                                    component={SignUp} 
                                />
                                <Route 
                                    path="/sign-in" 
                                    component={SignIn} 
                                />
                                <Route 
                                    path="/dashboard"
                                    component={Dashboard}
                                />
                            </Switch>
                        </ThemeProvider>
                    }
                </UserContext.Provider>
            </BrowserRouter>
        </>
    );
}

export default App;