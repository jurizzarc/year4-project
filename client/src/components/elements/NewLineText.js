import React from 'react';
import { Markup } from 'interweave';

const NewLineText = (props) => {
    const detectionType = props.detectionType;
    const detections = props.detections;
    const detectionsArray = detections.map(detection => detection.text);
    let extractedText = detectionsArray.join('');

    if (detectionType !== 'hndwrtng-img') {
        const newText = extractedText.split('\n').map((value, index) => {
            return (
                <p key={index}>
                    {value}
                </p>
            )
        });

        return newText;
    } else {
        // Pre Clean
        let cleanText = extractedText;
        ['[h]', '[s]', '[p]', '[b]', '[i]'].forEach((item, i) => {
            cleanText = cleanText.split(item).join('\n' + item.toUpperCase());
            cleanText = cleanText.split(item.toUpperCase()).join('\n' + item.toUpperCase());
        });
        cleanText = cleanText.split('\n\n').join('\n');

        let lines = cleanText.split('\n');
        let resultHtml = '';
        lines.forEach((line, i) => {
            let temp = line.substring(3);
            if (line.toUpperCase().includes('[H]')) {
                resultHtml = resultHtml + '<h1>' + temp + '</h1>';
            } else if (line.toUpperCase().includes('[S]')) {
                resultHtml = resultHtml + '<h2>' + temp + '</h2>';
            } else if (line.toUpperCase().includes('[P]')) {
                resultHtml = resultHtml + '<p>' + temp + '</p>';
            } else if (line.toUpperCase().includes('[B]')) {
                resultHtml = resultHtml + '<p><strong>' + temp + '</strong></p>';
            } else if (line.toUpperCase().includes('[I]')) {
                resultHtml = resultHtml + '<p><em>' + temp + '</em></p>'
            }
        });
        
        console.log(resultHtml);
        return <Markup content={resultHtml} />;
    }
}

export default NewLineText;