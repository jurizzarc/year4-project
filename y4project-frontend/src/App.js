// main component
import React, {useState} from "react";
import {BrowserRouter, Switch, Route} from "react-router-dom";
import Home from "./components/pages/Home";
import Header from "./components/layout/Header";
import Login from "./components/auth/Login";
import Register from "./components/auth/Register";
import UserContext from "./context/UserContext";

import "./style.css";

export default function App() {
    const [userData, setUserData] = useState({
        token: undefined,
        user: undefined
    });

    return <>
        <BrowserRouter>
            <UserContext.Provider value={{userData, setUserData}}>
                <Header />
                <Switch>
                    <Route exact path="/" component={Home} />
                    <Route path="/login" component={Login} />
                    <Route path="/register" component={Register} />
                </Switch>
            </UserContext.Provider>
        </BrowserRouter>
    </>;
}