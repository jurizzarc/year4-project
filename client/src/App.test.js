import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';
import NewLineText from './components/elements/NewLineText';

it('renders without crashing', () => {
    shallow(<App />);
});

const newLineTextProps = {
    detectedText: ['This is a handwritten text.'],
    detectionType: 'hndwrtng-img',
    hasHandWritingSystem: false
};
describe('', () => {
    it('accepts newLineText props', () => {
        const wrapper = mount(<NewLineText detections={newLineTextProps.detectedText} detectionType={newLineTextProps.detectionType} hasHandWritingSystem={newLineTextProps.hasHandWritingSystem} />);
        expect(wrapper.props().detections).toEqual(newLineTextProps.detectedText);
        expect(wrapper.props().detectionType).toEqual(newLineTextProps.detectionType);
        expect(wrapper.props().hasHandWritingSystem).toEqual(newLineTextProps.hasHandWritingSystem);
    });
});

it("renders correctly", () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
});