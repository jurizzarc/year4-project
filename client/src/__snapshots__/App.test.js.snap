// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <Memo(l) />
  <BrowserRouter>
    <ContextProvider
      value={
        Object {
          "setUserData": [Function],
          "userData": Object {
            "token": undefined,
            "user": undefined,
          },
        }
      }
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/sign-up"
        />
        <Route
          component={[Function]}
          path="/sign-in"
        />
        <Route
          component={[Function]}
          path="/dashboard"
        />
        <Route
          exact={true}
          path="/read/:userUploadId"
          render={[Function]}
        />
      </Switch>
    </ContextProvider>
  </BrowserRouter>
</Fragment>
`;
